### PR TITLE
CMR-5904

### PR DESCRIPTION
**Pull request by ygliuvt**
Changed get variable concepts by associations to batch searching instead of one association at a time.

### CODE REVIEW

File: indexer-app/src/cmr/indexer/data/concepts/variable.clj [hunk 1]
```diff
@@ -64,18 +64,17 @@
...
-      ;; elastic document for nested variables field
-      (assoc variable-concept :variable-association variable-association))))
+(defn- variable-associations->variable-concepts
+  "Returns the variable concepts for the given variable associations."
+  [context variable-associations]
+  (let [variable-concept-ids (map :variable-concept-id variable-associations)
+        id->association (zipmap variable-concept-ids variable-associations)
+        add-association-fn (fn [concept]
+                             (assoc concept :variable-association (get id->association (:concept-id concept))))
+        variable-concepts (mdb/get-latest-concepts context variable-concept-ids true)]
```
Comment (ygliuvt) [inline indexer-app/src/cmr/indexer/data/concepts/variable.clj]: I added `true` parameter to allow missing concepts. This is just defensive coding against unlikely scenario of corrupted database. In normal cases, every variable association should be linked to an existing variable concept.

**Comment by sxu123** [comment]
The ticket number should be 5094, not 5904 :)

**Comment by ygliuvt** [comment]
> The ticket number should be 5094, not 5904 :)

Thanks, I will update the commit message to indicate the correct ticket number.

File: indexer-app/src/cmr/indexer/data/concepts/service.clj [hunk 2]
```diff
@@ -59,16 +59,12 @@
...
-  "Returns the service concept and service association for the given service association."
-  [context service-association]
-  (let [{:keys [service-concept-id]} service-association
-        service-concept (mdb/find-latest-concept
-                         context
-                         {:concept-id service-concept-id}
-                         :service)]
-    (when-not (:deleted service-concept)
-      service-concept)))
+(defn- service-associations->service-concepts
```
Comment (oubiwann) [inline indexer-app/src/cmr/indexer/data/concepts/service.clj]: This is a big improvement, idiomatically too!

File: indexer-app/src/cmr/indexer/data/concepts/variable.clj (hunk 1 above)
Comment (oubiwann) [reply indexer-app/src/cmr/indexer/data/concepts/variable.clj]: I'm definitely against defensive programming -- however! We've been hit pretty hard in the past with data corruption issues, so I'm a big +1 on this :-)

File: indexer-app/src/cmr/indexer/data/concepts/variable.clj [hunk 3]
```diff
@@ -64,18 +64,17 @@
...
+    (->> variable-concepts
```
Comment (oubiwann) [inline indexer-app/src/cmr/indexer/data/concepts/variable.clj]: Again, very nice stylistic improvement here :-)

File: indexer-app/src/cmr/indexer/data/concepts/service.clj [hunk 4]
```diff
@@ -59,16 +59,12 @@
...
+  "Returns the service concepts for the given service associations."
+  [context service-associations]
+  (let [service-concept-ids (map :service-concept-id service-associations)
+        service-concepts (mdb/get-latest-concepts context service-concept-ids true)]
+    (filter #(not (:deleted %)) service-concepts)))
```
Comment (chris-durbin) [inline indexer-app/src/cmr/indexer/data/concepts/service.clj]: ```(remove :deleted service-concepts)```

File: indexer-app/src/cmr/indexer/data/concepts/variable.clj [hunk 5]
```diff
@@ -64,18 +64,17 @@
...
+         (filter #(not (:deleted %)))
```
Comment (chris-durbin) [inline indexer-app/src/cmr/indexer/data/concepts/variable.clj]: similar `(remove :deleted)`

**Pull request merged**